packages/metals-vscode/src/debugger/scalaDebugger.ts

```diff
@@ -79,7 +79,7 @@ async function runMain(main: ExtendedScalaRunMain): Promise<boolean> {
 
     const shellOptions = { ...platformSpecificOptions(), env };
     const task = new Task(
-      { type: "scala", task: "run" },
+      { type: "scala " + main.data.class, task: "run" },
```

Comment:
```suggestion
      { type: "scala", task: "run", class: main.data.class },
```
I read the API for Task and it seems `type` should stay as "scala" and we can just add another property which looks to be working. If we ever need to use https://code.visualstudio.com/api/extension-guides/task-provider#task-definition then we would avoid weird issue.

I should have thought of it in the first place, thanks for working on this!

Reply:
Done.